presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -926,10 +926,16 @@ public void finishDelete(Session session, TableHandle tableHandle, Collection<Sl
     @Override
     public Optional<ConnectorViewDefinition> getView(Session session, QualifiedObjectName viewName)
     {
-        return getOptionalCatalogMetadata(session, viewName.getCatalogName())
-                .flatMap(catalog -> catalog.getMetadata().getView(
-                        session.toConnectorSession(catalog.getCatalogName()),
-                        viewName.asSchemaTableName()));
+        Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, viewName.getCatalogName());
```

Comment:
```java
return getOptionalCatalogMetadata(session, viewName.getCatalogName())
        .flatMap(catalogMetadata -> {
            CatalogName catalogName = catalogMetadata.getConnectorId(session, viewName);
            ConnectorMetadata metadata = catalogMetadata.getMetadataFor(catalogName);
            ConnectorSession connectorSession = session.toConnectorSession(catalogName);
            return metadata.getView(connectorSession, viewName.asSchemaTableName());
        });

Reply:
Could be... I made the `getView` method look the same as `getTableHandle` method. The two methods do pretty much the same thing.

Also, `if (catalog.isPresent())` is proliferated within the class, so we may want to change all places at once (separate PR perhaps).